tests/core/scan_test.py

```diff
@@ -1,33 +1,77 @@
+import os
 import tempfile
 import textwrap
 
 import pytest
 
 from detect_secrets.core import scan
 from detect_secrets.settings import transient_settings
+from detect_secrets.util import git
+from detect_secrets.util.path import get_relative_path_if_in_cwd
 
 
-def test_handles_broken_yaml_gracefully():
-    with tempfile.NamedTemporaryFile(suffix='.yaml') as f:
-        f.write(
-            textwrap.dedent("""
-            metadata:
-                name: {{ .values.name }}
-            """)[1:].encode(),
+class TestGetFilesToScan:
+    @staticmethod
```

Comment:
Could we add a test that tries to call the function on a directory and a file at the same time?